src/PrivateRoute/index.jsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Route } from 'react-router-dom';
+import PropTypes from 'prop-types';
+
+import LoginRedirect from '../LoginRedirect';
+
+const PrivateRoute = ({
+  component: Component,
+  isAuthenticated,
+  path,
+  authenticatedAPIClient,
+  redirect,
+  ...rest
+}) => (
+  isAuthenticated === true
+    ? <Route
+      {...rest}
+      path={path}
+      component={Component}
+    />
+    : <Route
+      {...rest}
+      path={path}
+      render={props => (
+        <LoginRedirect
+          {...props}
+          redirect={redirect + path}
+          authenticatedAPIClient={authenticatedAPIClient}
+        />
+      )}
+    />
+);
+
+PrivateRoute.propTypes = {
+  isAuthenticated: PropTypes.bool.isRequired,
+  component: PropTypes.func.isRequired,
+  authenticatedAPIClient: PropTypes.shape({
+    login: PropTypes.func.isRequired,
+  }).isRequired,
+  redirect: PropTypes.string.isRequired,
+  location: PropTypes.shape({
+    state: PropTypes.shape({
+      from: PropTypes.shape({
+        pathname: PropTypes.string,
+      }),
+    }),
+  }).isRequired,
```

Comment:
This only makes location required, you probably want `isRequired` on `state` `from` and `pathname` as well.

Reply:
This was a copy/paste from existing `edx-portal` code. :) Do we need to define `location` at all here? It's not being used anywhere in our code.

Reply:
I believe it is defined here to reference the fact we expect these routes to be wrapped under a router that provides the location as props. You're just spreading props, so it shouldn't be needed. If you were using props.location you'd have to include it otherwise it would throw a warning